### ISSUE TITLE
[Bug] Too many requests on open Thread list

### ISSUE THREAD

**Issue by ankar84** [RocketChat/Rocket.Chat#22358]
### Description:

Sometimes users get `Error Too Many Requests` error when they just opens Thread list
I have default **API Rate Limiter** and **DDP Rate Limiter** settings

<details>

![image](https://user-images.githubusercontent.com/4023037/122019623-a4fd0580-cdcc-11eb-857f-1e086837cad9.png)
![image](https://user-images.githubusercontent.com/4023037/122019752-c958e200-cdcc-11eb-95d6-279e28b2bebc.png)

</details>

### Steps to reproduce:

1. Go to some chat (DM, private group or channel)
2. Click on Thread button 4 times
3. Get error Too Many Requests

### Expected behavior:

Thread list opens

### Actual behavior:

![thread-too-many](https://user-images.githubusercontent.com/4023037/122177905-b9a1d200-ce8e-11eb-9252-8d088bd2ab62.gif)

### Server Setup Information:

- Version of Rocket.Chat Server: 3.15.0
- Operating System: CentOS7
- Deployment Method: docker
- Number of Running Instances: 24
- DB Replicaset Oplog: Enabled
- NodeJS Version: 12.22.1
- MongoDB Version: 4.2.14 WiredTiger

### Client Setup Information

- Desktop App or Browser Version: 3.0.7
- Operating System: Windows 10

### Relevant logs:

```
Failed to load resource: the server responded with a status of 429 (Too Many Requests)
```

**Comment by mr-karan** [comment]
+1 Getting the same on `rocketchat/rocket.chat:3.15.0`

**Comment by mr-karan** [comment]
> > +1 Getting the same on `rocketchat/rocket.chat:3.15.0`
> 
> What client do you use? Web version? Desktop? If Desktop, then what version?
> Thanks!

Web version.

**Comment by ankar84** [comment]
> > > +1 Getting the same on `rocketchat/rocket.chat:3.15.0`
> > 
> > 
> > What client do you use? Web version? Desktop? If Desktop, then what version?
> > Thanks!
> 
> Web version.

Strange. I can't reproduce that in Web Version and in desktop version 3.2.2
But can reproduce that issue in Desktop version 3.0.7 with this error in console of DevTools
`Failed to load resource: the server responded with a status of 429 (Too Many Requests)`
![image](https://user-images.githubusercontent.com/4023037/122173684-af7dd480-ce8a-11eb-8f3d-2f1b8f3ac83a.png)

**Comment by rhnvrm** [comment]
I am able to reproduce this with opening thread button three times and opening it for the fourth time. RC seems to be sending 2 requests on clicking this.


![Screenshot_2021-06-16_12-32-35](https://user-images.githubusercontent.com/952036/122173160-1b6a3800-ce9f-11eb-82c5-eaaa22dc95e9.png)

Also, 

If you click on any thread, this call is made twice there as well. 
![Screenshot_2021-06-16_12-36-08](https://user-images.githubusercontent.com/952036/122173643-929fcc00-ce9f-11eb-81af-3073784b59c7.png)

**Comment by mr-karan** [comment]
Also, for additional context, DDP Rate Limiter shows no increase in metrics:

![image](https://user-images.githubusercontent.com/5689132/122173666-9cc1ca80-ce9f-11eb-81cf-82dbb5fbb83b.png)

**Comment by ankar84** [comment]
> RC seems to be sending 2 requests on clicking this.

Yeap, I see 2 requests on Thread list click
![thread-too-many](https://user-images.githubusercontent.com/4023037/122177905-b9a1d200-ce8e-11eb-9252-8d088bd2ab62.gif)

**Comment by ankar84** [comment]
But I'm still can reproduce that issue only in desktop version 3.0.7
Desktop version 3.2.2 and Web version not involved in my case.

**Comment by ankar84** [comment]
Disabling **API Rate Limiter** fixed that issue, but I not recommend that workaround
![image](https://user-images.githubusercontent.com/4023037/122710870-db71cf00-d269-11eb-96b8-de0f004d2b1c.png)

**Comment by ankar84** [comment]
Looks a bit different in 3.16.0
![thread02](https://user-images.githubusercontent.com/4023037/123614051-a5ee5800-d80c-11eb-8c8d-4889bb36ea53.gif)

**Comment by ankar84** [comment]
We are on 3.16.3 now and it looks like that issue is solver here.
I enabled back API Rate Limiter and test by steps to reproduce
![thread03](https://user-images.githubusercontent.com/4023037/126261062-f09d49cf-d9a9-4fa2-8cf3-7bb452efe5c8.gif)
Can anyone else check that and confirm that in solved in 3.16.3?

**Comment by ankar84** [comment]
Fixed in 3.16.3
